conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -60,17 +60,20 @@ class CMakeToolchain(object):
         {% macro iterate_configs(var_config, action) %}
             {% for it, values in var_config.items() %}
                 {% set genexpr = namespace(str='') %}
+                {% for conf, value in values -%}
+                set(CONAN_DEF_{{ it }} "{{ value }}")
```

Comment:
I had to separate variable definitions for generator expressions to support escaping, it was not working otherwise. Check: https://crascit.com/2022/01/25/quoting-in-cmake/